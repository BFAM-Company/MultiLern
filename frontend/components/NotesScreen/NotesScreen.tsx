import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import DrawingComponent from '../DrawingComponent/DrawingComponent';

function NotesScreen() {
  return (
      <View style={styles.container}>
        <DrawingComponent />
      </View>
  );
}

const styles = StyleSheet.create({
  container:{
    flex:1,
    width:'100%',
    alignItems:'center',
    justifyContent:'center'
  }
})

export default NotesScreen;