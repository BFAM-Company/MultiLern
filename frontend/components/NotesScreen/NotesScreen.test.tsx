import React from 'react';
import { render, cleanup, fireEvent, waitFor } from '@testing-library/react-native';
import NotesScreen from './NotesScreen';


jest.mock("@react-native-async-storage/async-storage", () => require("../../__mocks__/mock-async-storage"));

describe('NotesScreen', () => {
  afterEach(cleanup);
    it('renders correctly', () => {
        const mockPageSwitcher = jest.fn();
        const { toJSON } = render(<NotesScreen pageSwitcher={mockPageSwitcher} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
