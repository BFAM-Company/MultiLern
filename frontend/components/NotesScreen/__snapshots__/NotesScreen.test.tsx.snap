// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesScreen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(230,230,230)",
        "borderRadius": 20,
        "bottom": 10,
        "elevation": 16,
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-around",
        "left": 10,
        "overflow": "hidden",
        "position": "absolute",
        "right": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.44,
        "shadowRadius": 10.32,
        "width": "auto",
        "zIndex": 10000000,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderColor": "lightgray",
          "borderRadius": 100,
          "borderWidth": 4,
          "display": "flex",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
      testID="main-color-picker"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgray",
          "borderColor": "lightgray",
          "borderRadius": 100,
          "borderWidth": 4,
          "display": "flex",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#000",
            "borderRadius": 100,
            "height": 6,
            "width": 6,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgray",
          "borderColor": "lightgray",
          "borderRadius": 20,
          "borderWidth": 3,
          "display": "flex",
          "height": "auto",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 5,
          "width": "auto",
        }
      }
    >
      <Text
        style={
          {
            "color": "dimgray",
            "fontSize": 16,
            "fontWeight": "700",
            "margin": 5,
          }
        }
      >
        normal
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          [
            {
              "marginBottom": -20,
              "width": 30,
            },
            {
              "height": 140,
            },
          ]
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          [
            {
              "marginBottom": -20,
              "width": 30,
            },
            {
              "height": 100,
            },
          ]
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgray",
          "borderRadius": 100,
          "display": "flex",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Text
        style={
          {
            "color": "dimgray",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Aa
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "left": 16,
        "position": "absolute",
        "right": 10,
        "top": 50,
        "zIndex": 10000000,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": 30,
            "tintColor": "rgb(45,45,55)",
            "width": 30,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
